### ISSUE TITLE
Copy to/from matrix_base and eigen matrices

### ISSUE THREAD

**Issue by cdeterman** [viennacl/viennacl-dev#218]
I have been trying to look at the source code in `matrix.hpp` to figure out how to manually copy between `matrix_base` objects and `Eigen::Matrix`/`Eigen::Map<Eigen::Matrix>` objects and I am completely stuck.  Is there a current way using the current internal functions with ViennaCL or do some new functions need to be written?  I thought I could use something with the `memory_read` and `memory_write` functions but I am stumbling.  Any insight would be appreciated.

**Comment by cdeterman** [comment]
I have since gotten this to work by copying and modifying some of the other Eigen `copy` functions.  However this has led to other problems with functions such as `inplace_qr`.  I will continue to explore those problems but wanted to let you know the `copy` problem is no longer a major concern.  Ultimately all of this will be moot once issue #217 and issue #207 are solved so I can have unpadded matrices and interpret vectors as matrices.

**Comment by karlrupp** [comment]
yeah, currently only the copy to/from full ViennaCL matrices (no proxies) to/from Eigen matrices is supported. I recommend creating a full ViennaCL matrix as a temporary for the copies.
